File: backend/src/main/java/xyz/engsmyre/moviedescriptiongame/tmdb/service/MovieService.java
package xyz.engsmyre.moviedescriptiongame.tmdb.service;

import java.util.List;
import java.util.Random;
import org.springframework.stereotype.Service;
import xyz.engsmyre.moviedescriptiongame.tmdb.domain.Movie;
import xyz.engsmyre.moviedescriptiongame.tmdb.repository.MovieRepository;


@Service
public class MovieService {

    private final MovieRepository movieRepository;


    public MovieService(MovieRepository movieRepository) {
        this.movieRepository = movieRepository;
    }

    public Movie getRandomMovie() {     // TODO Add difficulty
        Random r = new Random();
        int pageCount = movieRepository.getPopularMoviesPageCount();
        int randomPage = r.nextInt(pageCount) + 1;
        List<Movie> randomPageMovies = movieRepository.getPopularMoviesFromPage(randomPage);
        int randomMovieIndex = r.nextInt(randomPageMovies.size());
        return randomPageMovies.get(randomMovieIndex);
    }

}

File: backend/src/main/kotlin/xyz/engsmyre/moviedescriptiongame/tmdb/service/MovieService.kt
package xyz.engsmyre.moviedescriptiongame.tmdb.service

import org.springframework.stereotype.Service
import xyz.engsmyre.moviedescriptiongame.tmdb.domain.Movie
import xyz.engsmyre.moviedescriptiongame.tmdb.repository.PopularMoviesRepository
import xyz.engsmyre.moviedescriptiongame.tmdb.repository.SearchableMovieRepository
import java.util.*
import kotlin.collections.ArrayList

@Service
class MovieService(private val popularMoviesRepository : PopularMoviesRepository,
                   private val searchableMoviesRepository: SearchableMovieRepository) {
    // TODO Add difficulty
    fun getRandomMovie() : Movie {//  TODO Add difficulty
        val r = Random()
        val pageCount = popularMoviesRepository.popularMoviesPageCount
        val randomPage = r.nextInt(pageCount) + 1
        val randomPageMovies = popularMoviesRepository.getPopularMoviesFromPage(randomPage)
        val randomMovieIndex = r.nextInt(randomPageMovies!!.size)
        return randomPageMovies[randomMovieIndex]!!
    }

    fun searchForMovies(movieQuery : String) : List<Movie> {
       return ArrayList<Movie>()
    }

}